server/src/main/java/org/elasticsearch/index/query/SpanOrQueryBuilder.java

```diff
@@ -115,14 +129,16 @@ public static SpanOrQueryBuilder fromXContent(XContentParser parser) throws IOEx
                         if (query instanceof SpanQueryBuilder == false) {
                             throw new ParsingException(parser.getTokenLocation(), "spanOr [clauses] must be of type span query");
                         }
-                        clauses.add((SpanQueryBuilder) query);
+                        final SpanQueryBuilder clause = (SpanQueryBuilder) query;
+                        checkBoostValue(parser.getTokenLocation(), clause);
+                        clauses.add(clause);
                     }
                 } else {
                     throw new ParsingException(parser.getTokenLocation(), "[span_or] query does not support [" + currentFieldName + "]");
                 }
             } else {
                 if (AbstractQueryBuilder.BOOST_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
-                    boost = parser.floatValue();
+                    throw new ParsingException(parser.getTokenLocation(), "spanOr [clauses] can't have boost value");
```

Comment:
That is great and I think the only necessary change in `SpanOrQueryBuilder`.  All other SpanQueryBuilders should have similar line in their `fromXContent` methods.